iothub/device/src/Transport/AmqpIoT/AmqpIoTReceivingLink.cs

```diff
@@ -25,40 +25,73 @@ internal class AmqpIoTReceivingLink
         public AmqpIoTReceivingLink(ReceivingAmqpLink receivingAmqpLink)
         {
             _receivingAmqpLink = receivingAmqpLink;
-            _receivingAmqpLink.Closed += _receivingAmqpLinkClosed;
+            _receivingAmqpLink.Closed += ReceivingAmqpLinkClosed;
         }
 
-        private void _receivingAmqpLinkClosed(object sender, EventArgs e)
+        private void ReceivingAmqpLinkClosed(object sender, EventArgs e)
         {
-            Closed.Invoke(sender, e);
+            if (Logging.IsEnabled) Logging.Enter(this, $"{nameof(ReceivingAmqpLinkClosed)}");
+            if (ReferenceEquals(sender, _receivingAmqpLink))
+            {
+                Closed?.Invoke(this, e);
+            }
+            else
+            {
+                if (Logging.IsEnabled) Logging.Error(this, "Amqp ReceivingLink closed event: instance mismatch.", $"{nameof(ReceivingAmqpLinkClosed)}");
```

Comment:
Isn't this a terminal case? (i.e. `Logging.Fault`?)

Reply:
It shouldn't happen so I removed `ReferenceEquals` check.

Reply:
missed it. Added now.